web/src/components/WithAnalytics/WithAnalytics.tsx

```diff
@@ -0,0 +1,16 @@
+import {useEffect} from 'react';
+import AnalyticsService from '../../services/Analytics/Analytics.service';
+
+const withAnalytics = <P extends object>(Component: React.ComponentType<P>, name: string) => {
```

Comment:
I see the value of the HOC, what's your opinion about `HOC` vs `custom hooks`? This could also be a hook at the router component level (maybe 🤔 ) that triggers the event each time the path changes.

Reply:
I'm on the fence in this one, I think we cannot do it at the router component as it would need to be wrapped by a `<BrowserRouter>` or `<Routes>` component to use the hooks to get the path.

Another thing is that we'll need to create a mapping to parse the paths to page names.

The HOC approach seems more explicit to me, but I agree that it might be overkill. I'm not sure 🤷🏽‍♀️